test/squid.js

```diff
@@ -0,0 +1,378 @@
+                amount,
+                true,
+                true,
+                1000000000000000,
+                CLOCK_PACKAGE_ID,
+            ],
+            typeArguments: [coins[coin1].type, coins[coin2].type],
+        });
+
+        await builder.signAndExecute(deployer);
+        /* place_limit_order<BaseAsset, QuoteAsset>(
+            self: &mut Pool<BaseAsset, QuoteAsset>,
+            balance_manager: &mut BalanceManager,
+            trade_proof: &TradeProof,
+            client_order_id: u64,
+            order_type: u8,
+            self_matching_option: u8,
+            price: u64,
+            quantity: u64,
+            is_bid: bool,
+            pay_with_deep: bool,
+            expire_timestamp: u64,
+            clock: &Clock,
+            ctx: &TxContext,
+        ) */
+    }
+
+    async function registerCoin(coin) {
+        const builder = new TxBuilder(client);
+
+        const coinInfo = await builder.moveCall({
+            target: `${deployments.its.packageId}::coin_info::from_metadata`,
+            arguments: [coins[coin].coinMetadata, 9],
+            typeArguments: [coins[coin].type],
+        });
+        const coinManagment = await builder.moveCall({
+            target: `${deployments.its.packageId}::coin_management::new_with_cap`,
+            arguments: [coins[coin].treasuryCap],
+            typeArguments: [coins[coin].type],
+        });
+        await builder.moveCall({
+            target: `${deployments.its.packageId}::its::register_coin`,
+            arguments: [objectIds.its, coinInfo, coinManagment],
+            typeArguments: [coins[coin].type],
+        });
+
+        const registerTxn = await builder.signAndExecute(deployer, { showEvents: true });
+
+        objectIds.tokenId = registerTxn.events[0].parsedJson.token_id.id;
+    }
+
+    before(async () => {
+        client = new SuiClient({ url: getFullnodeUrl(network) });
+
+        // Request funds from faucet
+        const addresses = [operator, deployer, keypair].map((keypair) => keypair.toSuiAddress());
+        await fundAccountsFromFaucet(addresses);
+
+        await deployDeepbook();
+
+        objectIds = {
+            balanceManager: await createBalanceManager(),
+            deepCoin: findObjectId(deployments.token.publishTxn, 'Coin'),
+            deepbookAdminCap: findObjectId(deployments.deepbook.publishTxn, 'DeepbookAdminCap'),
+            deepbookRegistry: findObjectId(deployments.deepbook.publishTxn, 'Registry', 'created', 'RegistryInner'),
+        };
+
+        dependencies.push('gas_service', 'example');
+
+        // Publish all packages
+        for (const packageDir of dependencies) {
+            const publishedReceipt = await publishPackage(client, deployer, packageDir, { showEvents: true });
+
+            deployments[packageDir] = publishedReceipt;
+        }
+
+        objectIds = {
+            ...objectIds,
+            squid: findObjectId(deployments.squid.publishTxn, 'squid::Squid'),
```

Comment:
Create a util `findObjects` that given the tx, and a list of types, returns a JS object with the key being the base type name, and value being the id. This should be useful for deployment scripts as well

Reply:
notice that at most two to three objects are obtained from each Tx, and the names need to be specified as well.